classes/image_types_ostree.bbclass

```diff
@@ -165,25 +165,19 @@ IMAGE_CMD_ostreecommit () {
     fi
 
     # Commit the result
-    ostree --repo=${OSTREE_REPO} commit \
+    ostree_target_hash=$(ostree --repo=${OSTREE_REPO} commit \
            --tree=dir=${OSTREE_ROOTFS} \
            --skip-if-unchanged \
            --branch=${OSTREE_BRANCHNAME} \
            --subject="${OSTREE_COMMIT_SUBJECT}" \
            --body="${OSTREE_COMMIT_BODY}" \
-           --add-metadata-string=version="${OSTREE_COMMIT_VERSION}" \
-           --bind-ref="${OSTREE_BRANCHNAME}-${IMAGE_BASENAME}"
+           --add-metadata-string=version="${OSTREE_COMMIT_VERSION}")
+
+    echo $ostree_target_hash > ${WORKDIR}/ostree_manifest
 
     if [ "${OSTREE_UPDATE_SUMMARY}" = "1" ]; then
         ostree --repo=${OSTREE_REPO} summary -u
     fi
-
-    # To enable simultaneous bitbaking of two images with the same branch name,
-    # create a new ref in the repo using the basename of the image. (This first
-    # requires deleting it if it already exists.) Fixes OTA-2211.
-    ostree --repo=${OSTREE_REPO} refs --delete ${OSTREE_BRANCHNAME}-${IMAGE_BASENAME}
-    ostree_target_hash=$(cat ${OSTREE_REPO}/refs/heads/${OSTREE_BRANCHNAME})
```

Comment:
I don't think I fully understand what was happening here. If another parallel bitbake also did commit before this line but after the line 168 was executed, then it will give a wrong hash, won't it?

Anyway, I think the current change makes this logic more straightforward.

Reply:
No, because this task is protected by a lockfile inside the repository location. So if multiple bitbake instances run at the same time, this code is guaranteed to run in a sequence.

https://github.com/advancedtelematic/meta-updater/blob/7a9dffea0efeb53d73a8f4065bdbf3b0638593c0/classes/image_types_ostree.bbclass#L161

Reply:
The `do_image_ota` task is not using that lock. So passing the checksum via the bitbake instance private TMPDIR guarantees that we do not end up using a different checksum in the `do_image_ota` task.